src/turnkeyml/analyze/util.py

```diff
@@ -100,6 +100,205 @@ def get_onnx_ops_list(onnx_model) -> Dict:
     return onnx_ops_counter
 
 
+def attribute_to_dict(attribute):
+    """
+    Helper function that returns a dictionary containing node attributes
+    """
+    attribute_dict = {}
+    for field in ["f", "i", "s"]:
+        if attribute.HasField(field):
+            attribute_dict[attribute.name] = getattr(attribute, field)
+            return attribute_dict
+    if attribute.ints:
+        attribute_dict[attribute.name] = list(attribute.ints)
+    elif attribute.floats:
+        attribute_dict[attribute.name] = list(attribute.floats)
+    elif attribute.strings:
+        attribute_dict[attribute.name] = list(attribute.strings)
+    else:
+        attribute_dict[attribute.name] = "unknown_type"
+    return attribute_dict
+
+
+def get_onnx_total_flops(onnx_model) -> Union[int, None]:
+    """
+    Calculate total number of FLOPs found in the onnx model.
+    FLOP is defined as one floating-point operation. This distinguishes
+    from multiply-accumulates (MACs) where FLOPs == 2 * MACs.
+    """
+    try:
+        model = onnx.shape_inference.infer_shapes(
+            onnx.load(onnx_model), strict_mode=True, data_prop=True
+        )
+    except Exception as e:  # pylint: disable=broad-except
+        printing.log_warning(f"Failed to get ONNX FLOPs from {onnx_model}: {str(e)}")
+        return None
+
+    # If the ONNX model contains one of the following unsupported ops, then this
+    # function will return None since the FLOP total is expected to be incorrect
+    unsupported_ops = [
+        "Einsum",
+        "RNN",
+        "GRU",
+        "DeformConv",
+    ]
```

Comment:
A thought about maintaining this code for the future: if the ONNX spec ever adds a new FLOP-contributing OP then we will need to add it to this list (or add a case for it below). I wonder if there is a good way to remind ourselves to do that? 

Or a good programmatic way to detect ops that emerged in a new opset that this function doesn't support yet? A blunt way would be "if model.opset > MAX_SUPPORTED_OPSET: return None". 

I'm not sure how common it is to get new FLOP-contributing ONNX ops these days. So this comment isn't a blocker for this PR, but something to think about nonetheless.